File: python/examples/providers/custom_api_key.py
from mirascope import llm

# Use a different API key for OpenAI
llm.register_provider("openai", api_key="sk-my-other-key")

# Or point to a different endpoint (e.g., a proxy)
llm.register_provider("openai", base_url="https://my-proxy.example.com/v1")

# Now all openai/ calls use the registered configuration
response = llm.use_model("openai/gpt-4o-mini").call("Say hello")
print(response.text())

File: typescript/examples/providers/custom-api-key.ts
/**
 * Custom API key example.
 *
 * Demonstrates registering a provider with a custom API key or base URL.
 *
 * Run with: bun run example examples/providers/custom-api-key.ts
 */
import { llm } from "mirascope";

// Use a different API key for OpenAI
llm.registerProvider("openai", { apiKey: "sk-my-other-key" });

// Or point to a different endpoint (e.g., a proxy)
llm.registerProvider("openai", { baseURL: "https://my-proxy.example.com/v1" });

// Now all openai/ calls use the registered configuration
const response = await llm.useModel("openai/gpt-4o-mini").call("Say hello");
console.log(response.text());
